tempodb/encoding/vparquet4/schema.go

```diff
@@ -0,0 +1,963 @@
+
+	// todo: remove this when support for tag based search is removed. we only
+	// need the below mappings for tag name search
+	labelMappings = map[string]string{
+		LabelRootSpanName:     "RootSpanName",
+		LabelRootServiceName:  "RootServiceName",
+		LabelServiceName:      "rs.list.element.Resource.ServiceName",
+		LabelCluster:          "rs.list.element.Resource.Cluster",
+		LabelNamespace:        "rs.list.element.Resource.Namespace",
+		LabelPod:              "rs.list.element.Resource.Pod",
+		LabelContainer:        "rs.list.element.Resource.Container",
+		LabelK8sClusterName:   "rs.list.element.Resource.K8sClusterName",
+		LabelK8sNamespaceName: "rs.list.element.Resource.K8sNamespaceName",
+		LabelK8sPodName:       "rs.list.element.Resource.K8sPodName",
+		LabelK8sContainerName: "rs.list.element.Resource.K8sContainerName",
+		LabelName:             "rs.list.element.ss.list.element.Spans.list.element.Name",
+		LabelHTTPMethod:       "rs.list.element.ss.list.element.Spans.list.element.HttpMethod",
+		LabelHTTPUrl:          "rs.list.element.ss.list.element.Spans.list.element.HttpUrl",
+		LabelHTTPStatusCode:   "rs.list.element.ss.list.element.Spans.list.element.HttpStatusCode",
+		LabelStatusCode:       "rs.list.element.ss.list.element.Spans.list.element.StatusCode",
+	}
+	// the two below are used in tag name search. they only include
+	//  custom attributes that are mapped to parquet "special" columns
+	traceqlResourceLabelMappings = map[string]string{
+		LabelServiceName:      "rs.list.element.Resource.ServiceName",
+		LabelCluster:          "rs.list.element.Resource.Cluster",
+		LabelNamespace:        "rs.list.element.Resource.Namespace",
+		LabelPod:              "rs.list.element.Resource.Pod",
+		LabelContainer:        "rs.list.element.Resource.Container",
+		LabelK8sClusterName:   "rs.list.element.Resource.K8sClusterName",
+		LabelK8sNamespaceName: "rs.list.element.Resource.K8sNamespaceName",
+		LabelK8sPodName:       "rs.list.element.Resource.K8sPodName",
+		LabelK8sContainerName: "rs.list.element.Resource.K8sContainerName",
+	}
+	traceqlSpanLabelMappings = map[string]string{
+		LabelHTTPMethod:     "rs.list.element.ss.list.element.Spans.list.element.HttpMethod",
+		LabelHTTPUrl:        "rs.list.element.ss.list.element.Spans.list.element.HttpUrl",
+		LabelHTTPStatusCode: "rs.list.element.ss.list.element.Spans.list.element.HttpStatusCode",
+	}
+
+	parquetSchema = parquet.SchemaOf(&Trace{})
+)
+
+// droppedAttrCounter represents an entity that can count dropped attributes
+type droppedAttrCounter interface {
+	addDroppedAttr(n int32)
+}
+
+type Attribute struct {
+	Key string `parquet:",snappy,dict"`
+
+	ValueType    AttrType  `parquet:",snappy,delta"`
+	Value        []string  `parquet:",snappy,dict,list"`
+	ValueInt     []int64   `parquet:",snappy,list"`
+	ValueDouble  []float64 `parquet:",snappy,list"`
+	ValueBool    []bool    `parquet:",snappy,list"`
+	ValueDropped string    `parquet:",snappy,optional"`
```

Comment:
quick comment on how this is encoded? json?

Reply:
Yes this this JSON encoded. The same encoding was used for the columns `ValueKVList` and `ValueArray` in vParquet3